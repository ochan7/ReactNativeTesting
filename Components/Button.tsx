import React, { useState } from 'react';
import { View, Button } from 'react-native';

export default function MyButton() {
  const [someText, updateSomeText] = useState('');
  return (
    <View>
      {someText}
      <Button
        title="Nothing to do for now"
        onPress={() => updateSomeText('CLICKED')}
      />
    </View>
  );
}
