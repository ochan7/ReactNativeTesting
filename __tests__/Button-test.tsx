import React from 'react';
import { create, act } from 'react-test-renderer';
import { MyButton } from '../Components';

describe('Button component', () => {
  test('it changes text when clicked', () => {
    let component: any;
    act(() => {
      component = create(<MyButton />);
    });
    if (!component) {
      throw new Error('Component cannot be found');
    }

    const instance = component.root;
    const wrapper = instance.findByType('View');
    let [text, button] = wrapper.children;

    expect(text).toEqual('');

    act(() => button.props.onPress());
    text = wrapper.children[0];
    expect(text).toEqual('CLICKED');
  });
});
