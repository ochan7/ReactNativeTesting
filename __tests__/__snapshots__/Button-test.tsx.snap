// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Nothing to do for now
    </Text>
  </View>
</View>
`;
